blog/2023-10-10-weaviate-1-22-release/index.mdx

```diff
@@ -0,0 +1,253 @@
+
+Future plans include the ability to index nested properties, for example to allow for filtering on nested properties and vectorization options.
+
+## Async vector indexing
+
+Weaviate now offers the option of asynchronous vector indexing to object creation. In turn, it enables you to optimize imports for speed where vector indexing is not required immediately.
+
+In the past, the HNSW index would be built synchronously with object creation. This meant that the import process would occur in lockstep with the indexing process. The benefit of the synchronous approach is that the user can perform a search immediately after import, as the data is already indexed. However, if your only goal was fastest data ingestion, this could slow down the import process.
+
+With `1.22`, the vector indexing process can now be performed asynchronously to object creation. This means that the import process can go on independently to the indexing process and continue to create objects while the indexing process is ongoing.
+
+Outwardly, this change does not affect the user experience, other than the import process returning a response faster than before.
+
+The user can still perform a search immediately after import, as Weaviate will search any un-indexed data using brute force. This means that the user will not have to wait for the indexing process to complete before being able to search the newly imported data.
+
+The `nodes` REST endpoint has been updated to reflect this change.
+
+`INDEXING` has been added to the list of available node statuses, and the `stats` field now includes an `vectorQueueLength` field, which contains the number of objects currently being indexed.
+
+As of `1.22`, this is an opt-in feature. It can be enabled by setting the `WEAVIATE_ASYNC_INDEXING` setting to `true`, such as in the `docker-compose.yml` file.
+
+## Additional gRPC support
+
+The team continues to add gRPC support under the hood, with a big part of the equivalent GraphQL `Get` queries, and batch import functionality now available via gRPC.
+
+With these changes, we have seen improvements in import speeds of around ~60-80% for batch imports, and multiple times faster `Get` queries. A big part of this comes from the fact that gRPC is a binary protocol, which is more efficient than the text-based GraphQL protocol, as well as it being based on HTTP/2.
+
+## New Python client API
+
+If you are wondering how to use the new gRPC functionality, we have you covered with the new Python client. This client is a pre-release version with not only gRPC support but also an improved API. The new API is more Pythonic, with a focus on improving the user/developer experience through custom classes and methods, which will trigger IDE auto-completion in place of untyped dictionaries.
```

Comment:
Consider

The new API is more Pythonic, with a focus on improving the user/developer experience. Custom classes and methods now trigger IDE auto-completion in place of untyped dictionaries.